src/wp-includes/class-wp-network.php

```diff
@@ -353,17 +353,13 @@ public static function get_by_path( $domain = '', $path = '', $segments = null )
 		 */
 		$using_paths = true;
 		if ( wp_using_ext_object_cache() ) {
```

Comment:
This line confuses me, is the idea to avoid an expensive query unless there is a persistent cache to make it worthwhile? 

Not a blocker because it appears to have been around for a long time.

Reply:
> is the idea to avoid an expensive query unless there is a persistent cache to make it worthwhile?

Yes, exactly. If the cache is not persistent it will run on every page load and isn't worth doing.